AppCenterCrashes/AppCenterCrashes/MSCrashes.mm

```diff
@@ -273,6 +297,12 @@ - (instancetype)init {
 - (void)applyEnabledState:(BOOL)isEnabled {
   [super applyEnabledState:isEnabled];
 
+#if !TARGET_OS_OSX
```

Comment:
Why is it here? Should we only remove the observer when it is being disabled?

Reply:
It's the same pattern as here
https://github.com/microsoft/appcenter-sdk-apple/blob/4bea6348f2f0b4c51898c86d50302e59888931f1/AppCenter/AppCenter/MSAppCenter.m#L553-L556

Reply:
It isn't consistent. I can see this in `MSAuth`.
https://github.com/microsoft/appcenter-sdk-apple/blob/develop/AppCenterAuth/AppCenterAuth/MSAuth.m#L126

Reply:
This isn't a big deal but it doesn't need to remove the observer when it is being enabled. I don't want to block this PR because of this one but we should revisit this one in `MSAppCenter`.